tests.lua

```diff
@@ -145,6 +145,30 @@ function LibSerialize:RunTests()
     end
 
 
+    --[[---------------------------------------------------------------------------
+        Test of async serialization
+    --]]---------------------------------------------------------------------------
+    do
+        local t1 = { x = "y", "test", [false] = { 1, 2, 3, a = "b" } }
+        local opts = {
+            stable = true,
+            filter = function(t, k, v) return not tonumber(k) or tonumber(k) < 100 end
+        }
+        local serialized1 = LibSerialize:SerializeEx(opts, t1)
+        opts.async = true
+        -- setup serialize coroutine
+        local thread = coroutine.create(LibSerialize.SerializeEx)
```

Comment:
It's kind of a pain to require the user to do this. What if we moved it into `SerializeEx` and changed the return in that case? For example:

```lua
local state = LibSerialize:SerializeEx({ async = true }, myTable);
while not state.done do
    -- obviously this wouldn't be done in a loop but you get the idea
    state:continue()
end
return state.result
```

Reply:
This would also be good to add to the examples section above and in the readme, on how to correctly use the option.